### PR TITLE
Fix num literals in YAML scalar block highlighting

### PR DESCRIPTION
**Pull request by ashmaroli**
Standalone num literals in a block scalar should be tokenized as `Literal::Number` not `Str`.
However if those literals are preceded or succeeded by other non-whitespace characters, then it would be a legit `Str` token.
The exception is if the next tokens are `whitespace` and a `comment`:

```yaml
# Mapping scalars to scalars
---
hr:  65    # Home runs
avg: 0.278 # Batting average
rbi: 147   # Runs Batted In
time: 20:03:20
staff:
  - name: Mark McGwire
    hr:   65
    avg:  0.278
    addr: B 221 Block
```
YAML Visual Test

### CODE REVIEW

File: lib/rouge/lexers/yaml.rb [hunk 1]
```diff
@@ -328,7 +328,7 @@ def set_indent(match, opts={})
       state :plain_scalar_in_block_context do
         # the : indicator ends a scalar
         rule %r/[ ]*(?=:[ \n]|:$)/, Text, :pop!
-        rule %r/[ ]*:/, Str
+        rule %r/[ ]*:\S+/, Str
```
Comment (pyrmont) [inline lib/rouge/lexers/yaml.rb]: I don't follow the significance of `\S+` here. Is this related to the number issue as well?

Comment (ashmaroli) [reply lib/rouge/lexers/yaml.rb]: Oh looks like I forgot to note this in the opening comment..
The deal here is to handle the datetime strings such as `time: 20:03:20`. Without that `\S`, the match would end just after the colon. But we want the match to proceed till the scanner encounters a whitespace character.

Comment (pyrmont) [reply lib/rouge/lexers/yaml.rb]: Ahhh... make sense.

Comment (pyrmont) [reply lib/rouge/lexers/yaml.rb]: Oops, I did have an additional follow-up.

When I check the display of YAML on the current master in the visual testing app, I don't see any problems in the highlighting of the various time values that are in the sample. Was there a specific form that was causing an issue?

Comment (ashmaroli) [reply lib/rouge/lexers/yaml.rb]: Once you add just `rule %r/\d+(?:\.\d+)?(?=(\r?\n)| +#)/, Literal::Number, :pop!` to `ln#342` on `master`, you'll notice the discrepancy with the time strings.

Comment (pyrmont) [reply lib/rouge/lexers/yaml.rb]: Now it's all clear. Thanks!
